pkg/providers/imagefamily/image.go

```diff
@@ -51,11 +52,11 @@ const (
 	imageIDFormat = "/CommunityGalleries/%s/images/%s/versions/%s"
 )
 
-func NewProvider(kubernetesInterface kubernetes.Interface, kubernetesVersionCache *cache.Cache, versionsClient CommunityGalleryImageVersionsAPI, location string) *Provider {
+func NewProvider(opts *options.Options, kubernetesInterface kubernetes.Interface, kubernetesVersionCache *cache.Cache, versionsClient CommunityGalleryImageVersionsAPI) *Provider {
 	return &Provider{
 		kubernetesVersionCache: kubernetesVersionCache,
 		imageCache:             cache.New(imageExpirationInterval, imageCacheCleaningInterval),
-		location:               location,
+		location:               opts.Location,
```

Comment:
why not just put options in the Provider struct itself and we can access location from that? I also will need some values from options in some of these providers in this pr https://github.com/Azure/karpenter-provider-azure/pull/365/files so that pattern would make access a bit easier.

Reply:
(discussed offline)

I am still hesitate to make these providers depends on `options` entirely. 
Right now it does module-wise, but not the struct, as the only reference to `options` is on creation function (`NewProvider`). Maybe the better name to this is `NewProviderFromOptions`.

But again, not a strong opinion. Wondering what do others think? Should we be more extreme on one side (e.g., using `opts`/`ctx` everywhere)?